src/ashley/models.py

```diff
@@ -189,6 +190,30 @@ def get_user_roles(self, user) -> List[str]:
         roles = [group.replace(role_group_prefix, "") for group in user_role_groups]
         return roles
 
+    def get_course_info(self):
+        """Retrieve course info in the LTI identifier of the context.
+
+        Returns
+        -------
+        dict
+            school_name: the school name
+            course_name: the course name
+            course_section: the course section
+
+        """
+        # Extract the course informations if the LTI id of this context is in
+        # the OpenEdx format
+        openedx_match = re.match(r"^course-v[0-9]:(.*)$", self.lti_id)
+        if openedx_match:
+            part = openedx_match.group(1).split("+")
+            length = len(part)
+            return {
+                "school_name": part[0] if length >= 1 else None,
+                "course_name": part[1] if length >= 2 else None,
+                "course_run": part[2] if length >= 3 else None,
```

Comment:
Shouldn't we raise an error if not all fields are filled?

Reply:
To be honest, I used the same logic than in Marsha here.

In this case, I'm not comfortable with raising an error when the outside world does not reflect our expectations.
I see it as a "best-effort" data extraction function.

Reply:
IMO we should be pedantic on this: I would prefer raising an error and fix it if data is missing than emit (and store) a partial/corrupted record (or thousands of them) that will never be used.

The same paradigm applies for Marsha. Ping @lunika :wink:

Reply:
As discussed, these information could be contained in a parent in the context as followed.

```
context_activities=ContextActivities(
      parent=[
          Activity(
              id=<course_key>,
              definition=ActivityDefinition(
                  name=LanguageMap({"en-US": "test course"}),
                  type="http://adlnet.gov/expapi/activities/course",
              ),
          )
      ]
  )
```
The triplet {school, course, session} would be gathered in the `course_key` (ID of the parent)

Reply:
Thanks! I added a fixup commit to address this.

Note that the course name is not included, we don't store it in the current state of the project.
(And, for example, OpenEdx dogwood does not send it in the LTI request)